src/xjadeo/homography.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * homography.h
+ *
+ *  based on code from arturo castro
```

Comment:
What license does this code have? Can it be used in terms of the GPL?

Reply:
Arturo castro is one of the developers of OpenFrameworks, he posted the code this is based on in the OF forum, so i asume it has the license of OpenFrameworks (Mit license)

Reply:
Could you verify that and add the license boilerplate?
MIT is fine (compatible with xjadeo's GPL)

Reply:
I found it on [github](https://github.com/arturoc/homography.h) and it has GNU Affero General Public License v3.0, so i respected that and added reference to the repository, along the one to the forum that also was there.

Reply:
AGPL is a problem.

The code could be **linked** with xjadeo's GPLv2 after upgrading the project to GPL version 3, but using it as compile-time header is not allowed.

Reply:
Didn't know about that, i have to review these new versions of the GPL. So, this code cannot be included because Xjadeo doesn't use AGPL? Then if i understand correctly to include this code i should have to make it into a library that has to be distributed under AGPL3.0 and then it can be included only in AGPL projects but could be linked by GPL3.0 projects.

Its a bit messi, and then, that library could not be distributed with a GPL3 project, should be hosted has a separate project? i am correct?

Perhaps i should try finding another source for homography calculation, i know it is included in different open source projects, (opencv to say one.. should check license).

Any suggestion?

Reply:
I'm asking around. it _might_ be possible to just  use it as C file, compile it into a dedicated .o  object and link it.

Since the rest of xjadeo is GPLv2+  then product would then be AGPLv3 when configured with --enable-warp.

Reply:
Alas, it's not so easy :(

 A conversion on #gnu IRC points out that the resulting binary cannot be distributed:

> if you modify the gplv2 work to include the agplv3 work, then you already cannot distribute the result.
>
> To take advantage of the limited compatibility between agplv3 and gplv3 you need to upgrade. otherwise you simply don't have the option. Copyleft licenses are by nature mutually incompatible with each other.

This would mean that all of xjadeo would have to be explicitly relicensed under the GPLv3  first.

Also it was noted that the code `findHomography` has some references to code copied from
"Multiple View Geometry in Computer Vision 2ed Hartley R. and Zisserman A."  That is (C) and unlikely to be in terms of the AGPL.

I guess you're fine for the art-project, but eventually we'll have to re-do the homography math.